include/compiler.h

```diff
@@ -0,0 +1,26 @@
+/* compiler.h - A secure memory allocator
+ * Copyright 2022 - chris.rohlf@gmail.com */
+
+#pragma once
+
+#define INTERNAL_HIDDEN __attribute__((visibility("hidden")))
+#define ASSUME_ALIGNED __attribute__((assume_aligned(8)))
+#define CONST __attribute__((const))
+
+/* This isn't standard in C as [[nodiscard]] until C23 */
```

Comment:
We should document the minimum required C version to use/compile isoalloc I think.